pkg/authorization/rewrite/rewrite.go

```diff
@@ -63,7 +43,14 @@ type rewritingAuthorizer struct {
 
 var _ authorizer.Authorizer = &rewritingAuthorizer{}
 
-// GetRequestAttributes populates authorizer attributes for the requests to kube-rbac-proxy.
+// Authorize implements the authorizer.Authorizer interface.
+// It generates a list of attributes based on the given attributes and context.
+// It then calls the delegate authorizer with each of the generated attributes.
+// If any of the generated attributes is not authorized, the request won't be authorized,
+// it follows an AND logic.
+// If no attributes get generated, the request won't be authorized. This can only
+// happen if the rewrites logic is being used and the client didn't provide any values
+// in the header or query.
```

Comment:
`Authorize()` does not implement the `authorizer.Authorizer` interface, `rewritingAuthorizer` does that. This `Authorize()` is but a means of the `*rewritingAuthorizer` to implement the interface.

Explain what the function does not how it does it, long comments are evil.